### PR TITLE
A lot of small stuff

### PR DESCRIPTION
**Pull request by Soapy7261**
Make py-cord install extra packages for speed, perform LF normalization, make paw understand who send its own messages, don't reverse the messages array, more type hinting, use states instead of bools which allows to check if we've stopped because of the token limit

### CODE REVIEW

**Comment by Soapy7261** [comment]
Wait i forgot to push 1 other thing

**Comment by Soapy7261** [comment]
If paw starts mis-understanding context, just flip `oldest_first` to `False` in socials.py line 261

File: data.py [hunk 1]
```diff
@@ -168,4 +168,5 @@
 You have access to the chat history of the current chat.
 Use the conversation history as context for your prompts.
 If asked about what happened in earlier messages, use the chat history given to you.
+Do not under any circumstances say 'Generating...', no exceptions.
```
Comment (ToothyDev) [inline data.py:171]: Did this happen in testing, or why add this specifically?

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Auto detect text files and perform LF normalization
+* text=auto
```
Comment (ToothyDev) [inline .gitattributes:2]: What does this exactly do (for the bot)

File: cogs/socials.py [hunk 3]
```diff
@@ -255,15 +255,25 @@ async def gay(self, ctx, user=None, border=False, server_avatar=True):
     @slash_command()
     @option("text", str, description="What do you want to tell Paw?")
     # @commands.cooldown(1, 30, commands.BucketType.user)
-    async def gpt(self, ctx: discord.ApplicationContext, text):
+    async def gpt(self, ctx: discord.ApplicationContext, text: str):
         """ Talk to Paw! """
-        messages = await ctx.channel.history(limit=50).flatten()
         await ctx.defer()
-        messages.reverse()
+        messages = await ctx.channel.history(limit=50, oldest_first=True).flatten()
```
Comment (ToothyDev) [inline cogs/socials.py]: The part you deleted is needed no matter what. I just tested and regardless of using oldest_First=True or =False, the messages are not in chronological order. I don't know why it is like that, but that is exactly why I just reversed it
Just remove the oldest_first from this and keep the messages.reverse(), if you print each message as it gets appended you can see that it has all kinds of orders, but never a chronological one

**Review by ToothyDev — CHANGES_REQUESTED** [review]

File: data.py (hunk 1 above)
Comment (Soapy7261) [reply data.py:171]: Check the gpt code for why this is, line 272

File: .gitattributes (hunk 2 above)
Comment (Soapy7261) [reply .gitattributes:2]: This makes it so if your on windows and clone the repo it clones it in CRLF but if your on unix it clones it in LF, not needed but nice for system compat, if what im told is correct

File: cogs/socials.py (hunk 3 above)
Comment (Soapy7261) [reply cogs/socials.py]: Oh

**Review by ToothyDev — APPROVED** [review]

**Pull request merged**